src/Juvix/Core/Parameterisations/Naturals.hs

```diff
@@ -0,0 +1,38 @@
+module Juvix.Core.Parameterisations.Naturals where
+
+import Juvix.Core.Types hiding (apply, parseTy, parseVal, reservedNames, reservedOpNames, typeOf)
+import Juvix.Library
+import Text.ParserCombinators.Parsec
+import qualified Text.ParserCombinators.Parsec.Token as Token
+import Prelude (String)
+
+data NatTy
+  = Nat
+  deriving (Show, Eq)
+
+data NatVal
+  = Natural Natural
+  deriving (Show, Eq)
+
+typeOf ∷ NatVal → NatTy
+typeOf (Natural _) = Nat
+
+apply ∷ NatVal → NatVal → Maybe NatVal
+apply _ _ = Nothing
+
+parseTy ∷ Token.GenTokenParser String () Identity → Parser NatTy
+parseTy lexer = do
+  Token.reserved lexer "Nat"
+  pure Nat
+
+parseVal ∷ Token.GenTokenParser String () Identity → Parser NatVal
+parseVal lexer = Natural . fromIntegral |<< Token.natural lexer
+
+reservedNames ∷ [String]
+reservedNames = ["Nat"]
+
+reservedOpNames ∷ [String]
+reservedOpNames = []
+
+naturals ∷ Parameterisation NatTy NatVal
+naturals = Parameterisation typeOf apply parseTy parseVal reservedNames reservedOpNames
```

Comment:
Seems like a fine way of doing parameterization, What else could we parameterize over here?

Reply:
In the future this will include backend-specific translations, e.g. for LLVM.